x-pack/legacy/plugins/monitoring/public/lib/setup_mode.js

```diff
@@ -30,6 +30,15 @@ const setupModeState = {
 
 export const getSetupModeState = () => setupModeState;
 
+export const setNewlyDiscoveredClusterUuid = clusterUuid => {
+  const globalState = angularState.injector.get('globalState');
+  const executor = angularState.injector.get('$executor');
+  globalState.cluster_uuid = clusterUuid;
+  globalState.save();
+  angularState.scope.$apply();
```

Comment:
Maybe worth wrapping it in a safe handler instead? eg:
```javascript
angularState.scope.$apply(() => (globalState.cluster_uuid = clusterUuid));
```

Reply:
Would that work? We'd still need to call `.save()` though? What are the benefits of this approach?

Reply:
It's more optimal in that it'll only digest scope on what ever expression is passed, and I'll wrap it in try/catch incase there is already a digest in-progress: https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$apply

You can put `globalState.save();` in the $apply handler as well, if you think it has any bindings to update